torch/_inductor/codegen/cpp_micro_gemm.py

```diff
@@ -517,24 +517,8 @@ class CppMicroGemmAMX(CppMicroGemm):
     const auto num_elements_per_b_tile = 512;
     const auto last_k_offset = K / {{block_k}} * {{block_k}};
     const auto tail_k_size = K - last_k_offset;
-    const auto buf_size_per_nr_block = ((K / {{block_k}}) * num_elements_per_b_tile + tail_k_size * 32) * 2;
-    // Although the outer loop in the micro-kernel iterates on n, we could only cache K * Nr elements of
-    // dequantized B, pertaining to each value of n.
-    // But we allow for caching of multiple K * Nr sets in the code to support N != Nr for two reasons:
-    // 1. If N would not be equal to Nr in the future, outer loop in the micro-kernel that iterates over n
-    //    could potentially be unrolled. If the cache could only store K * Nr elements, loop-unrolling would be
-    //    impossible due to memory aliasing.
-    // 2. In the future, if the loops that iterate over m & n would need to be inverted,
-    //    such that m would be iterated over in the outer-loop instead, then for correctness,
-    //    we would need multiple dequantized B caches of K * Nr elements, if N would not be equal to Nr,
-    //    but would instead be a multiple of it.
-    // However, if performance would be better with iterating over n in the outer-loop,
-    // then we may revisit this decision because in that case, we could choose K & N such that
-    // K * Nr elements would fit in L1D but K * N elements need not (Other aspects of blocking
-    // would also be considered while determining the values of K & N).
-    // Then we could only cache one set of K * Nr elements instead of multiple sets of K * Nr elements.
-    // Of course, loop unrolling for the outer loop (iterating over n) would not be possible in that case.
-    const auto buf_size = buf_size_per_nr_block * (N / {{block_n}});
+    // we cache K * {{block_n}} elements of dequantized B, each sized 2 bytes
+    const auto buf_size = K * {{block_n}} * 2;
```

Comment:
nit: can we avoid hardcode `2` here but use data type sizes?

Reply:
Sure, thanks!
`sizeof({{input2_t}})` returns incorrect value, so I could try something else